lessons/beginners/list/index.md

```diff
@@ -0,0 +1,567 @@
+totiž *nový* seznam.
+Bude mít sice stejné prvky ve stejném pořadí,
+ale nebude to ten samý seznam:
+měnit se bude nezávisle na tom starém.
+
+```python
+a = [1, 2, 3]
+b = list(a)
+
+print(b)
+a.append(4)
+print(b)
+```
+
+Další způsob, jak tvořit seznamy
+(zvláště složitější), je nejdřív udělat prázdný
+seznam, a pak ho postupně naplnit pomocí funkce `append`.
+Třeba pokud z nějakého důvodu chceš seznam
+mocnin dvou, projdi čísla kterými chceme mocnit
+cyklem `for`, a pro každé z nich
+do seznamu přidej příslušnou mocninu:
+
+```python
+mocniny_dvou = []
+for cislo in range(10):
+    mocniny_dvou.append(2 ** cislo)
+print(mocniny_dvou)
+```
+
+Chceš-li seznam, který reprezentuje balíček karet,
+zavolej `append` pro všechny kombinace barev a hodnot.
+
+```python
+balicek = []
+for barva in '♠', '♥', '♦', '♣':  # (Na Windows použij textová jména)
+    for hodnota in list(range(2, 11)) + ['J', 'Q', 'K', 'A']:
+        balicek.append(str(hodnota) + barva)
+print(balicek)
+```
+
+## Seznamy a řetězce
+
+Seznamy a řetězce jsou druhy „sekvencí”,
+takže snad nepřekvapí, že se dá různě převádět
+z jednoho typu na druhý.
+Funkce `list` vytvoří z řetězce
+seznam znaků.
+Když chceme dostat seznam slov, použijeme
+na řetězci metodu `split` (angl. *rozdělit*):
+
+```python
+slova = 'Tato věta je složitá, rozdělme ji na slova!'.split()
+print(slova)
+```
+
+Metoda `split` umí brát i argument.
+Pokud ho předáme, místo mezer (a nových řádků)
+se řetězec „rozseká” daným oddělovačem.
+Takže když máme nějaká data oddělená čárkami,
+není nic jednoduššího než použít `split` s čárkou:
+
+```python
+zaznamy = '3A,8B,2E,9D'.split(',')
+print(zaznamy)
+```
+
+Chceme-li spojit seznam řetězců zase dohromady,
+do jediného řetězce, použijeme metodu
+`join` (angl. *spojit*).
+Pozor, tahle metoda se volá na *oddělovači*,
+tedy řetězci, kterým se jednotlivé kousky „slepí”
+dohromady; a jako argument bere seznam jednotlivých
+řetězců.
+
+```python
+veta = ' '.join(slova)
+print(veta)
+```
+
+## Úkol
+
+Někte{{gnd('ří', 'ré', both='ří')}} z vás pro 1-D piškvorky udělali funkci
+`tah_pocitace`, která umí hrát jen za křížky.
+Takové funkce se nedají přímo použít v turnaji,
+kde jeden z hráčů musí hrát za kolečka.
+
+Jedno z možných řešení je napsat funkci, která zamění v řetězci všechna `'x'`
+za `'o'~ a naopak.
+
+```python
+print(zamen_xo('---xo--xooxxox-'))  # → '---ox--oxxooxo-'
+```
+
+Pak se dá napsat funkce, která pokud má hrát
+za kolečka,
+
+* zamění v hracím poli 'x'↔'o',
+* na výsledek zavolá funkci, která zahraje křížek,
+* a ve výsledku opět zamění 'x'↔'o'.
+
+Původní křížky a kolečka tak zůstanou na svých
+místech, ale nově přidaný křížek se nahradí za kolečko.
+
+```python
+def tah_pocitace(pole, symbol):
+    if symbol == 'x':
+        return tah_pocitace_x(pole)
+    else:
+        return zamen_xo(tah_pocitace_x(zamen_xo(pole)))
+```
+
+Zkus napsat funkci <code>zamen_xo</code>.
+Udělej v ní seznam jednotlivých znaků (viz sekce Tvoření seznamů),
```

Comment:
Jo, je na to [funkce anchor](https://github.com/pyvec/naucse.python.cz/blame/b46a98cac74ded1f1cbdffbd1b17395abd724de9/lessons/beginners/install/linux.md#L81).